pkg/sloop/storemanager/storemanager.go

```diff
@@ -191,34 +196,77 @@ func doCleanup(tables typed.Tables, timeLimit time.Duration, sizeLimitBytes int,
+	}
+
+	beforeGCTime := time.Now()
+	for cleanUpTimeCondition(minPartition, maxPartition, timeLimit) || numOfKeysToDeleteForFileSizeCondition > 0 {
 		partStart, partEnd, err := untyped.GetTimeRangeForPartition(minPartition)
 		glog.Infof("GC removing partition %q with data from %v to %v (err %v)", minPartition, partStart, partEnd, err)
 		var errMessages []string
 		for _, tableName := range tables.GetTableNames() {
 			prefix := fmt.Sprintf("/%s/%s", tableName, minPartition)
 			start := time.Now()
-			err, numOfDeletedKeys, numOfKeysToDelete := untyped.DeleteKeysWithPrefix([]byte(prefix), tables.Db(), deletionBatchSize)
-			metricGcDeletedNumberOfKeysByTable.WithLabelValues(fmt.Sprintf("%v", tableName)).Set(numOfDeletedKeys)
-			totalNumOfDeletedKeys += numOfDeletedKeys
-			totalNumOfKeysToDelete += numOfKeysToDelete
+			err, numOfDeletedKeysforPrefix, numOfKeysToDeleteForPrefix := common.DeleteKeysWithPrefix([]byte(prefix), tables.Db(), deletionBatchSize)
+			metricGcDeletedNumberOfKeysByTable.WithLabelValues(fmt.Sprintf("%v", tableName)).Set(numOfDeletedKeysforPrefix)
+			totalNumOfDeletedKeys += numOfDeletedKeysforPrefix
+			totalNumOfKeysToDelete += numOfKeysToDeleteForPrefix
 			elapsed := time.Since(start)
-			glog.Infof("Call to DeleteKeysWithPrefix(%v) took %v and removed %f keys with error: %v", prefix, elapsed, numOfDeletedKeys, err)
+			glog.Infof("Call to DeleteKeysWithPrefix(%v) took %v and removed %f keys with error: %v", prefix, elapsed, numOfDeletedKeysforPrefix, err)
 			if err != nil {
 				errMessages = append(errMessages, fmt.Sprintf("failed to cleanup with min key: %s, elapsed: %v,err: %v,", prefix, elapsed, err))
 			}
 			anyCleanupPerformed = true
 		}
 
+		minPartitionStartTime = minPartitionStartTime.Add(1 * untyped.GetPartitionDuration())
+		minPartition = untyped.GetPartitionId(minPartitionStartTime)
+
+		minPartitionAge, err := untyped.GetAgeOfPartitionInHours(minPartition)
+		if err == nil {
+			metricAgeOfMinimumPartition.Set(minPartitionAge)
+		}
+
+		if minPartitionAge < 0 {
+			return false, totalNumOfDeletedKeys, totalNumOfKeysToDelete, fmt.Errorf("minimun partition age cannot be less than zero")
+		}
+
 		if len(errMessages) != 0 {
 			var errMsg string
 			for _, er := range errMessages {
 				errMsg += er + ","
 			}
 			return false, totalNumOfDeletedKeys, totalNumOfKeysToDelete, fmt.Errorf(errMsg)
 		}
+
+		glog.Infof("Deleted Number of keys so far: %v ", totalNumOfDeletedKeys)
+		if numOfKeysToDeleteForFileSizeCondition > totalNumOfDeletedKeys {
+			numOfKeysToDeleteForFileSizeCondition -= totalNumOfDeletedKeys
+			glog.Infof("Remaining number of keys to delete: %v  ", numOfKeysToDeleteForFileSizeCondition)
+		} else {
+			// Deleted number of keys is greater or equal. We have reached the required deletion.
+			numOfKeysToDeleteForFileSizeCondition = 0
+			glog.Infof("Remaining number of keys to delete: %v  ", numOfKeysToDeleteForFileSizeCondition)
```

Comment:
consider moving the log statement below the if-else statements - the log statement on line 258 seems odd after setting `numOfKeysToDeleteForFileSizeCondition` to 0